### PR TITLE
Added support for local document drive state and operations

### LINKED ISSUE #25: Integrate Drive Settings
[Notion Link](https://www.notion.so/makerdao-ses/CDN-1-09-Drive-Settings-status-to-from-local-cloud-available-offline-69d232f627334a47897293d4df41db98?pvs=4)

## **User Story**

As a Connect user, I want to be able to change the status of my drives between local, Secure cloud storage, and available offline, so that I can manage how and where my files are stored.

## Requirements
- **Drive Functionality Update**:
    - Changing to **Local**: The drive should become an offline storage on the user's device. The user is prompted that files are no longer available for shared access unless moved to the shared drive.
    - Changing to **Cloud (Shared Drive)**: The drive is now part of the collaborative cloud space. The user is prompted that files can now be collaboratively edited and published to the MakerDAO Atlas.
    - **Available Offline**: The drive should cache its contents for offline access. A notification informs the user that the drive's contents are now accessible offline, but collaborative edits won't sync until back online.

**Depends on: https://github.com/powerhouse-inc/design-system/issues/101**

https://www.figma.com/embed?embed_host=share&url=https%3A%2F%2Fwww.figma.com%2Ffile%2FnixRuw4mg4v8URY9rk1dTp%2FPH-(Connect-App)%3Ftype%3Ddesign%26node-id%3D5418%253A21080%26mode%3Ddesign%26t%3DgIrXrXzfJg4kTYjh-1

### PR DESCRIPTION
**Pull request by acaldas**
Fixes #25 and #41

### CODE REVIEW

File: src/components/drive-container.tsx [hunk 1]
```diff
@@ -129,20 +129,37 @@ export default function DriveContainer(props: DriveContainerProps) {
...
-            onItemOptionsClick={onItemOptionsClick}
-            onSubmitInput={item => onSubmitInput(item)}
-            onCancelInput={cancelInputHandler}
-            onDragStart={onDragStartHandler}
-            onDragEnd={onDragEndHandler}
-            onDropEvent={onDropEventHandler}
-            onDropActivate={onDropActivateHandler}
-            disableHighlightStyles={disableHoverStyles}
-        />
+        <>
```
Comment (ryanwolhuter) [inline src/components/drive-container.tsx:134]: did you leave this commented out here on purpose?

File: src/hooks/useDrivesContainer.ts [hunk 2]
```diff
@@ -2,12 +2,17 @@ import {
     BaseTreeItem,
     DriveViewProps,
     TreeItem,
+    SharingType as TreeItemSharingType,
```
Comment (ryanwolhuter) [inline src/hooks/useDrivesContainer.ts:5]: would you prefer us to rename this type to be more specific like this in the design system code?

**Review by ryanwolhuter — APPROVED** [review]
nice

File: src/hooks/useDrivesContainer.ts (hunk 2 above)
Comment (acaldas) [reply src/hooks/useDrivesContainer.ts:5]: I think the best fix is to change the sharing type in the document model to also be uppercase.

File: src/components/drive-container.tsx (hunk 1 above)
Comment (acaldas) [reply src/components/drive-container.tsx:134]: I'll hardcode some drives for the demo and then I'll cleanup before merging this pR

**Review by gpuente — APPROVED** [review]

**Comment by acaldas** [comment]
Depends on https://github.com/powerhouse-inc/design-system/pull/133

**Pull request merged**